desktop_version/src/GraphicsResources.cpp

```diff
@@ -47,69 +44,258 @@ SDL_Surface* LoadImage(const char *filename)
+    return optimizedImage;
+}
+
+/* Can't be static, used in Screen.h */
+SDL_Surface* LoadImageSurface(const char* filename)
+{
+    unsigned char* data;
+
+    SDL_Surface* loadedImage = LoadImageRaw(filename, &data);
+
+    SDL_Surface* optimizedImage = LoadSurfaceFromRaw(loadedImage);
     if (loadedImage != NULL)
     {
-        optimizedImage = SDL_ConvertSurfaceFormat(
-            loadedImage,
-            SDL_PIXELFORMAT_ARGB8888,
-            0
-        );
         VVV_freefunc(SDL_FreeSurface, loadedImage);
-        VVV_free(data);
-        SDL_SetSurfaceBlendMode(optimizedImage, SDL_BLENDMODE_BLEND);
-        return optimizedImage;
     }
-    else
+
+    VVV_free(data);
+
+    if (optimizedImage == NULL)
     {
         VVV_free(data);
         vlog_error("Image not found: %s", filename);
         SDL_assert(0 && "Image not found! See stderr.");
-        return NULL;
     }
+
+    return optimizedImage;
+}
+
+static SDL_Texture* LoadTextureFromRaw(const char* filename, SDL_Surface* loadedImage, const TextureLoadType loadtype)
+{
+    if (loadedImage != NULL)
+    {
+        // Modify the surface with the load type.
+        // This could be done in LoadImageRaw, however currently, surfaces are only used for
+        // pixel perfect collision (which will be changed later,) and the window icon.
+
+        switch (loadtype)
+        {
+        case TEX_WHITE:
+            SDL_LockSurface(loadedImage);
+            for (int i = 0; i < loadedImage->w * loadedImage->h; i++)
+            {
+                Uint8 r, g, b, a;
+                SDL_GetRGBA(((Uint32*)loadedImage->pixels)[i], loadedImage->format, &r, &g, &b, &a);
+                ((Uint32*)loadedImage->pixels)[i] = SDL_MapRGBA(loadedImage->format, 255, 255, 255, a);
+            }
+            SDL_UnlockSurface(loadedImage);
+            break;
+        case TEX_GRAYSCALE:
+            SDL_LockSurface(loadedImage);
+            for (int i = 0; i < loadedImage->w * loadedImage->h; i++)
+            {
+                Uint8 r, g, b, a;
+                SDL_GetRGBA(((Uint32*)loadedImage->pixels)[i], loadedImage->format, &r, &g, &b, &a);
+
+                // Magic numbers used for grayscaling (eyes percieve certain colors brighter than others)
+                r *= 0.299;
+                g *= 0.587;
+                b *= 0.114;
+
+                const double gray = SDL_floor(r + g + b + 0.5);
+
+                ((Uint32*)loadedImage->pixels)[i] = SDL_MapRGBA(loadedImage->format, gray, gray, gray, a);
+            }
+            SDL_UnlockSurface(loadedImage);
+            break;
+        default:
+            break;
+        }
+
+        //Create texture from surface pixels
+        SDL_Texture* texture = SDL_CreateTextureFromSurface(gameScreen.m_renderer, loadedImage);
+        if (texture == NULL)
+        {
+            vlog_error("Failed creating texture: %s. SDL error: %s\n", filename, SDL_GetError());
+        }
+
+        return texture;
+    }
+
+    return NULL;
```

Comment:
If you're enclosing the entire function in an if-statement, then you should move it out of the if-statement and negate the conditional. Especially if the conditional is a sanity check like making sure the argument isn't `NULL`.

Secondly, you should use the `DrawPixel` and `ReadPixel` functions from `GraphicsUtil.cpp` instead of re-implementing the logic here. (`DrawPixel` will need to be un-`static`ed and exported in `GraphicsUtil.h`.)

Also:
- Put a space between a cast and the expression it's casting.
- You misspelled "perceive".
- You forgot a space between a comment starter `//` and the text of the comment.
- You ended a parenthetical in a comment with a comma ("`(which will be changed later,)`").